src/app/webui/components/hooks/useChat.ts

```
@@ -0,0 +1,249 @@
+
+const generateUniqueId = () => `msg-${Date.now()}-${Math.random().toString(36).substring(2, 9)}`;
+
+export function useChat(wsUrl: string) {
+    const wsRef = useRef<globalThis.WebSocket | null>(null);
+    const [messages, setMessages] = useState<Message[]>([]);
+    // Store the last toolResult image URI to attach to the next AI response
+    const lastImageUriRef = useRef<string | null>(null);
+    const [status, setStatus] = useState<'connecting' | 'open' | 'closed'>('connecting');
+
+    useEffect(() => {
+        const ws = new WebSocket(wsUrl);
```

ESLint

[error] 'WebSocket' is not defined.

(no-undef)